Source/Core/DolphinQt/Config/ConfigDialog.cpp

```diff
@@ -0,0 +1,391 @@
+	for (const std::string& backend : AudioCommon::GetSoundBackends())
+		m_ui->cmbAudioBackend->insertItem(m_ui->cmbAudioBackend->count(), QString::fromStdString(backend));
+
+	InitStaticData();
+	LoadSettings();
+	SetupSlots();
+}
+DConfigDialog::~DConfigDialog()
+{
+}
+
+void DConfigDialog::UpdateIcons()
+{
+	m_ui->actionPageGeneral->setIcon(Resources::GetIcon(Resources::TOOLBAR_CONFIGURE));
+	m_ui->actionPageGraphics->setIcon(Resources::GetIcon(Resources::TOOLBAR_GRAPHICS));
+	m_ui->actionPageAudio->setIcon(Resources::GetIcon(Resources::TOOLBAR_AUDIO));
+	m_ui->actionPageControllers->setIcon(Resources::GetIcon(Resources::TOOLBAR_CONTROLLERS));
+}
+
+// Static data (translation-specific mappings to enums, etc.)
+static QMap<int, QString> s_cpu_engines;
+static QMap<TEXIDevices, QString> s_exi_devices;
+
+void DConfigDialog::InitStaticData()
+{
+	s_cpu_engines = {
+		 { PowerPC::CORE_INTERPRETER, tr("Interpreter (slowest)") },
+		 { PowerPC::CORE_CACHEDINTERPRETER, tr("Cached Interpreter (slower)") },
+		 #ifdef _M_X86_64
+		 { PowerPC::CORE_JIT64, tr("JIT Recompiler (recommended)") },
+		 { PowerPC::CORE_JITIL64, tr("JITIL Recompiler (slow, experimental)") },
+		 #elif defined(_M_ARM_64)
+		 { PowerPC::CORE_JITARM64, tr("JIT Arm64 (experimental)") },
+		 #endif
+	};
+	m_ui->cmbCpuEngine->clear();
+	m_ui->cmbCpuEngine->insertItems(0, s_cpu_engines.values());
+
+	s_exi_devices = {
+		{ EXIDEVICE_NONE, tr("<Nothing>") },
+		{ EXIDEVICE_DUMMY, tr("Dummy") },
+
+		{ EXIDEVICE_MEMORYCARD, tr("Memory Card") },
+		{ EXIDEVICE_MEMORYCARDFOLDER, tr("GCI Folder") },
+		{ EXIDEVICE_MIC, tr("Mic") },
+		{ EXIDEVICE_ETH, QStringLiteral("BBA") },
+		{ EXIDEVICE_AGP, QStringLiteral("Advance Game Port") },
+		{ EXIDEVICE_AM_BASEBOARD, tr("AM-Baseboard") },
+		{ EXIDEVICE_GECKO, QStringLiteral("USBGecko") }
+	};
+	m_ui->cmbGCSlotA->clear();
+	m_ui->cmbGCSlotA->insertItems(0, s_exi_devices.values());
+	m_ui->cmbGCSlotB->clear();
+	m_ui->cmbGCSlotB->insertItems(0, s_exi_devices.values());
+	m_ui->cmbGCSP1->clear();
+	m_ui->cmbGCSP1->insertItems(0, { tr("<Nothing>"), tr("Dummy"),
+		QStringLiteral("BBA"), tr("AM-Baseboard") });
+}
+
+void DConfigDialog::SetupSlots()
+{
+	// Helper macros for signal/slot creation
+#define SCGI SConfig::GetInstance()
+#define cAction(ACTION, CALLBACK) \
+	connect(m_ui->ACTION, &QAction::triggered, [this]() -> void CALLBACK)
+#define cCombo(COMBO, CALLBACK) \
+	connect(m_ui->COMBO, &QComboBox::currentTextChanged, [this]() -> void CALLBACK)
+#define cCheck(CHECKBOX, CALLBACK) \
+	connect(m_ui->CHECKBOX, &QCheckBox::stateChanged, [this]() -> void CALLBACK)
+#define cGbCheck(CHECKBOX, CALLBACK) \
+	connect(m_ui->CHECKBOX, &QGroupBox::toggled, [this]() -> void CALLBACK)
+#define cSpin(SPINBOX, CALLBACK) \
+	connect(m_ui->SPINBOX, static_cast<void(QSpinBox::*)(int)>(&QSpinBox::valueChanged), [this]() -> void CALLBACK)
+#define cSlider(SLIDER, CALLBACK) \
+	connect(m_ui->SLIDER, &QSlider::valueChanged, [this]() -> void CALLBACK)
```

Comment:
Please make these explicit, instead of using the preprocessor as a copy-paste mechanism (just make GCC output the preprocessed file with `gcc -E` or something, if you want to save time amending this)